### PR TITLE
feat: apply theme from sense

### PR DESCRIPTION
**Pull request by LiKang6688**
Screenshot 2021-12-31 at 02 05 41

### CODE REVIEW

File: src/table/components/TableHeadWrapper.jsx [hunk 1]
```diff
@@ -45,8 +46,11 @@ function TableHeadWrapper({
   focusedCellCoord,
   setFocusedCellCoord,
   keyboard,
+  backgroundColor,
 }) {
   const headStyle = useMemo(() => getHeadStyle(layout, theme), [layout, theme.name()]);
+  // transparent is not allowed
+  headStyle.backgroundColor = backgroundColor === 'transparent' ? 'inherent' : backgroundColor;
```
Comment (haxxmaxx) [inline src/table/components/TableHeadWrapper.jsx:53]: This should be done in `getHeadStyle`

File: src/table/components/TableWrapper.jsx [hunk 2]
```diff
@@ -50,14 +50,19 @@ export default function TableWrapper(props) {
...
   const { size, rows, columns } = tableData;
   const { page, rowsPerPage, rowsPerPageOptions } = pageInfo;
   const [focusedCellCoord, setFocusedCellCoord] = useState([0, 0]);
   const shouldRefocus = useRef(false);
   const tableContainerRef = useRef();
   const tableWrapperRef = useRef();
-  const classes = useStyles();
+
+  const backgroundColorFromTheme = theme.getStyle('object', 'straightTable', 'backgroundColor');
+  const backgroundColor = !backgroundColorFromTheme ? 'inherit' : backgroundColorFromTheme;
```
Comment (haxxmaxx) [inline src/table/components/TableWrapper.jsx:63]:
```suggestion
  const backgroundColor = theme.getStyle('object', 'straightTable', 'backgroundColor') || 'inherit';
```

File: src/table/components/TableHeadWrapper.jsx (hunk 1 above)
Comment (haxxmaxx) [inline src/table/components/TableHeadWrapper.jsx:53]: Is this only for the header? Can you have transparent on the rest of the table?

File: src/table/components/TableWrapper.jsx [hunk 3]
```diff
@@ -50,14 +50,19 @@ export default function TableWrapper(props) {
...
   } = props;
   const { size, rows, columns } = tableData;
   const { page, rowsPerPage, rowsPerPageOptions } = pageInfo;
   const [focusedCellCoord, setFocusedCellCoord] = useState([0, 0]);
   const shouldRefocus = useRef(false);
   const tableContainerRef = useRef();
   const tableWrapperRef = useRef();
-  const classes = useStyles();
+
+  const backgroundColorFromTheme = theme.getStyle('object', 'straightTable', 'backgroundColor');
```
Comment (haxxmaxx) [inline src/table/components/TableWrapper.jsx:62]: Unless `'straightTable'` is used for view data we might want to consider having a separate property for this table. Name TBD in that case I suppose

File: src/table/components/TableHeadWrapper.jsx (hunk 1 above)
Comment (LiKang6688) [reply src/table/components/TableHeadWrapper.jsx:53]: yes, we can have transparent on the rest of the table

Comment (haxxmaxx) [reply src/table/components/TableHeadWrapper.jsx:53]: why not the header?

**Comment by haxxmaxx** [comment]
what would the equivalent old table look like? For view data this might worsen the user experience, since the user can't control the font colors. In the image you can barely see the text on the black background

**Comment by LiKang6688** [comment]
A new PR to replace this one will come soon

**Pull request closed without merging**